src/rest_vol_attr.c

```diff
@@ -2698,6 +2874,36 @@ RV_attr_specific(void *obj, const H5VL_loc_params_t *loc_params, H5VL_attr_speci
             FUNC_DONE_ERROR(H5E_ATTR, H5E_CANTCLOSEOBJ, FAIL, "invalid attribute parent object type");
     } /* end if */
 
+    if (attr)
+        if (RV_attr_close(attr, H5P_DEFAULT, NULL) < 0)
+            FUNC_DONE_ERROR(H5E_ATTR, H5E_CANTCLOSEOBJ, FAIL, "can't close attribute");
+
+    if (renamed_attr)
+        if (RV_attr_close(renamed_attr, H5P_DEFAULT, NULL) < 0)
+            FUNC_DONE_ERROR(H5E_ATTR, H5E_CANTCLOSEOBJ, FAIL, "can't close attribute");
+
+    if (attr_parent)
+        switch (parent_obj_type) {
+            case H5I_FILE:
+            case H5I_GROUP:
+                if (RV_group_close((void *)attr_parent, H5P_DEFAULT, NULL) < 0)
+                    FUNC_GOTO_ERROR(H5E_SYM, H5E_CANTCLOSEOBJ, FAIL, "can't close parent group");
```

Comment:
Should be FUNC_DONE_ERROR here and below so we don't potentially cause an infinite loop